python/ray/serve/deployment_state.py

```diff
@@ -1506,16 +1506,34 @@ def _check_and_update_replicas(self) -> bool:
                     f"Resources required for each replica: {required}, "
                     f"resources available: {available}."
                 )
+                logger.warning(message)
                 if _SCALING_LOG_ENABLED:
                     print_verbose_scaling_log()
+                # If status is UNHEALTHY, give it higher priority over the stuck
```

Comment:
I think we can expand on this comment and its reasoning bit. Here's a suggestion:

```suggestion
                # If status is UNHEALTHY, leave the message as is. The issue that caused the deployment to be unhealthy should be prioritized over this resource availability issue.
```

This is a bit malformatted since it's one big line, so I recommend copying it into your local code and running the formatting script on it. Feel free to edit this comment further.